readthedocs/proxito/views/serve.py

```diff
@@ -362,10 +364,28 @@ def get(self, request, proxito_path, template_name='404.html'):
                     )
                     resp = HttpResponse(build_media_storage.open(storage_filename_path).read())
                     resp.status_code = 404
+                    self._register_broken_link(
+                        project=final_project,
+                        version=version,
+                        path=path,
+                    )
                     return resp
 
+        self._register_broken_link(
+            project=final_project,
+            version=version,
+            path=path,
```

Comment:
There is case where the path would be `/` instead of  `/en/notfound/` if there isn't a version named `notfound`, probably we should check for that before saving it.

Reply:
I don't fully follow, but we should definitely record the exact URL, however we can. Can we just use `request.get_full_path` or similar?

Reply:
yeah, I'm also +1 on saving the full path https://github.com/readthedocs/readthedocs.org/pull/9027#discussion_r829562056. But we still need the path that doesn't have the lang/version for the normal page views, so we can group them by path.